commands/migrate.go

```diff
@@ -0,0 +1,143 @@
+package commands
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/fnproject/cli/common"
+	"github.com/fnproject/cli/config"
+	yamltojson "github.com/ghodss/yaml"
+	"github.com/mitchellh/mapstructure"
+	"github.com/urfave/cli"
+	yaml "gopkg.in/yaml.v2"
+)
+
+type migrateFnCmd struct {
+	ff   *common.FuncFile
+	ffV2 *common.FuncFileV2
+}
+
+func MigrateCommand() cli.Command {
+	m := &migrateFnCmd{ff: &common.FuncFile{}, ffV2: &common.FuncFileV2{}}
+
+	return cli.Command{
+		Name:        "migrate",
+		Usage:       "Migrate a local func.yaml file to the new version",
+		Category:    "DEVELOPMENT COMMANDS",
+		Aliases:     []string{"m"},
+		Description: "Migrate will detect the version of the current func.yaml file, to the new version.",
```

Comment:
"This command will detect the version of a func.yaml file and update it to match the latest version supported by the Fn CLI.  Any old or unsupported attributes will be removed, and any new ones may be added."